purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1048,31 +1048,40 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
             appInBackground,
             { offeringsJSON ->
                 try {
-                    val jsonArrayOfOfferings = offeringsJSON.getJSONArray("offerings")
-                    val skus = mutableSetOf<String>()
-                    for (i in 0 until jsonArrayOfOfferings.length()) {
-                        val jsonPackagesArray =
-                            jsonArrayOfOfferings.getJSONObject(i).getJSONArray("packages")
-                        for (j in 0 until jsonPackagesArray.length()) {
-                            skus.add(
-                                jsonPackagesArray.getJSONObject(j)
-                                    .getString("platform_product_identifier")
-                            )
-                        }
+                    val skus = extractSkus(offeringsJSON)
+                    if (skus.isEmpty()) {
+                        handleErrorFetchingOfferings(
+                            PurchasesError(
+                                PurchasesErrorCode.ConfigurationError,
+                                OfferingStrings.CONFIGURATION_ERROR_NO_PRODUCTS_FOR_OFFERINGS
+                            ),
+                            completion
+                        )
+                    } else {
+                        getSkuDetails(skus, { detailsByID ->
+                            val offerings = offeringsJSON.createOfferings(detailsByID)
+
+                            if (offerings.all.isEmpty()) {
+                                handleErrorFetchingOfferings(
+                                    PurchasesError(
+                                        PurchasesErrorCode.ConfigurationError,
+                                        OfferingStrings.CONFIGURATION_ERROR_PRODUCTS_NOT_FOUND
+                                    ),
+                                    completion
+                                )
+                            } else {
+                                logMissingProducts(offerings, detailsByID)
```

Comment:
should we log missing products when empty too? or somehow else pass in which products were not found?

Reply:
Oh yeah that makes sense, we do that in iOS.